### PR TITLE
fix: store id committment as little endian

### PR DESCRIPTION
**Pull request by chaitanyaprem**
# Description
ID committment was being stored as big endian and hence tree roots are different when compared to nwaku. This fixes and stores it as little endian.

Tested and compare with nwaku to ensure root are same.

Note: The TODO will be fixed in a separate issue.

### CODE REVIEW

File: waku/v2/protocol/rln/group_manager/dynamic/dynamic.go [hunk 1]
```diff
@@ -225,7 +235,11 @@ func (gm *DynamicGroupManager) InsertMembers(toInsert *om.OrderedMap) error {
 			if oldestIndexInBlock == nil {
 				oldestIndexInBlock = evt.Index
 			}
-			idCommitments = append(idCommitments, rln.BigIntToBytes32(evt.IdCommitment))
+			convertedIdCommittment := rln.BigIntToBytes32(evt.IdCommitment)
+			leIdCommittment := ConvertToLittleEndian(convertedIdCommittment[:])
+			idCommitments = append(idCommitments, leIdCommittment)
```
Comment (richard-ramos) [inline waku/v2/protocol/rln/group_manager/dynamic/dynamic.go:240]: `rln.BigIntToBytes32` already uses little endian. https://github.com/waku-org/go-zerokit-rln/blob/master/rln/utils.go#L87-L92

How do you see this difference? Were proofs failing? I wonder if it's a case of display differences while internally the merkle root is correct

Comment (richard-ramos) [reply waku/v2/protocol/rln/group_manager/dynamic/dynamic.go:240]: Do note that go's big.Int `Bytes()` uses BE, and BigIntToBytes32 reverses the bytes

Comment (chaitanyaprem) [reply waku/v2/protocol/rln/group_manager/dynamic/dynamic.go:240]: > `rln.BigIntToBytes32` already uses little endian. https://github.com/waku-org/go-zerokit-rln/blob/master/rln/utils.go#L87-L92
> 
> How do you see this difference? Were proofs failing? I wonder if it's a case of display differences while internally the merkle root is correct

Interesting.
Me and Aryamann was debugging a different issue and noticed that merkle roots were different between go-waku and nwaku

Then we started digging deeper by printing id commitments..and noticed that before my change..the printed commitment was in big endian after the conversion using the function you had mentioned. 
After I made this change, the I'd commitment was printed properly and the first root matches in stored roots. 

I did not verify if the proof verification is failing or not. 

Will test it once without the change and with the change to confirm

Comment (chaitanyaprem) [reply waku/v2/protocol/rln/group_manager/dynamic/dynamic.go:240]: Retesting with code from master, proof seems to be valid. 
One more observation is when i test using this code and try to send a message, i notice the below logs in local node itself which indicates failed validation.

```

2023-12-27 17:00:27 2023-12-27T11:30:27.917Z    DEBUG   basichost       basic/basic_host.go:341 failed to fetch local IPv6 address      {"node": "16Uiu2HAmKoVrfYHpNGXJg82XWB8bE8qbwTxbvouXWErCWzs51AJp", "error": "no route found for ::"}
2023-12-27 17:00:28 2023-12-27T11:30:28.179Z    DEBUG   gowaku.node2    rln/waku_rln_relay.go:242       rln-relay topic validator called        {"node": "16Uiu2HAmKoVrfYHpNGXJg82XWB8bE8qbwTxbvouXWErCWzs51AJp", "hash": "0x3c69c36d2defef9abb1dfdf4b079273d694032b1bc8f10eb2fc97daba2e902c0", "pubsubTopic": "/waku/2/rs/1/7", "contentTopic": "/prem/1/abc/proto"}
2023-12-27 17:00:28 2023-12-27T11:30:28.179Z    DEBUG   gowaku.node2    rln/waku_rln_relay.go:141       invalid message: unexpected root        {"node": "16Uiu2HAmKoVrfYHpNGXJg82XWB8bE8qbwTxbvouXWErCWzs51AJp", "msgRoot": "0x8347b2181caf6c59e9e24484498d4114aadc7063cbc306592d737a26dcda3c14"}
2023-12-27 17:00:28 2023-12-27T11:30:28.179Z    DEBUG   gowaku.node2    rln/waku_rln_relay.go:258       message could not be verified   {"node": "16Uiu2HAmKoVrfYHpNGXJg82XWB8bE8qbwTxbvouXWErCWzs51AJp", "hash": "0x3c69c36d2defef9abb1dfdf4b079273d694032b1bc8f10eb2fc97daba2e902c0", "pubsubTopic": "/waku/2/rs/1/7", "contentTopic": "/prem/1/abc/proto"}
2023-12-27 17:00:28 2023-12-27T11:30:28.179Z    DEBUG   pubsub  go-libp2p-pubsub@v0.9.3/validation.go:476       validation done; took 110.5µs   {"node": "16Uiu2HAmKoVrfYHpNGXJg82XWB8bE8qbwTxbvouXWErCWzs51AJp"}
2023-12-27 17:00:28 2023-12-27T11:30:28.179Z    DEBUG   pubsub  go-libp2p-pubsub@v0.9.3/validation.go:338       message validation failed; dropping message from 16Uiu2HAmKoVrfYHpNGXJg82XWB8bE8qbwTxbvouXWErCWzs51AJp  {"node": "16Uiu2HAmKoVrfYHpNGXJg82XWB8bE8qbwTxbvouXWErCWzs51AJp"}
2023-12-27 17:00:28 2023-12-27T11:30:28.179Z    ERROR   gowaku.relay    rest/relay.go:195       publishing message      {"node": "16Uiu2HAmKoVrfYHpNGXJg82XWB8bE8qbwTxbvouXWErCWzs51AJp", "error": "validation failed"} 
```
As you rightly pointed out, maybe it was just a print issue then.

Comment (chaitanyaprem) [reply waku/v2/protocol/rln/group_manager/dynamic/dynamic.go:240]: Comparing RLN first root with a nwaku node confirms the roots also match when using go-waku master branch. 
Looks like subsequent roots being different reported by rln-db-inspector caused this confusion. 
Hence i will close this PR.

**Comment by chaitanyaprem** [comment]
As discussed [here](https://github.com/waku-org/go-waku/pull/946#discussion_r1417299974), this change is not required and hence closing the PR

**Pull request closed without merging**